core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/udf/java/JavaUDFOpDesc.scala

```diff
@@ -109,23 +122,40 @@ class JavaUDFOpDesc extends LogicalOp {
       Map(operatorInfo.outputPorts.head.id -> outputSchema)
     }
 
-    if (workers > 1)
-      PhysicalOp
-        .oneToOnePhysicalOp(
-          workflowId,
-          executionId,
-          operatorIdentifier,
-          OpExecWithCode(code, "java")
-        )
-        .withDerivePartition(_ => UnknownPartition())
-        .withInputPorts(operatorInfo.inputPorts)
-        .withOutputPorts(operatorInfo.outputPorts)
-        .withPartitionRequirement(partitionRequirement)
-        .withIsOneToManyOp(true)
-        .withParallelizable(true)
-        .withSuggestedWorkerNum(workers)
-        .withPropagateSchema(SchemaPropagationFunc(propagateSchema))
-    else
+    if (parallelizable) {
+      if (advanced) {
+        PhysicalOp
+          .oneToOnePhysicalOp(
+            workflowId,
+            executionId,
+            operatorIdentifier,
+            OpExecWithCode(code, "java")
+          )
+          .withDerivePartition(_ => UnknownPartition())
+          .withInputPorts(operatorInfo.inputPorts)
+          .withOutputPorts(operatorInfo.outputPorts)
+          .withPartitionRequirement(partitionRequirement)
+          .withIsOneToManyOp(true)
+          .withParallelizable(true)
+          .withSuggestedWorkerNum(workers)
+          .withPropagateSchema(SchemaPropagationFunc(propagateSchema))
+      } else {
+        PhysicalOp
+          .oneToOnePhysicalOp(
+            workflowId,
+            executionId,
+            operatorIdentifier,
+            OpExecWithCode(code, "java")
+          )
+          .withDerivePartition(_ => UnknownPartition())
+          .withInputPorts(operatorInfo.inputPorts)
+          .withOutputPorts(operatorInfo.outputPorts)
```

Comment:
merge the common code. only apply a difference part (i.e.,  `.withParallelizable(true)`) to different cases

Reply:
see pythonUDFSourceOpDescV2 for example.